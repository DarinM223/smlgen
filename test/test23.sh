../smlgen test23.sml no:s --recurmod --test
if cmp --silent test23.sml.test test23.sml.expected; then
  echo "test23 succeeded"
else
  echo "test23 failed"
  echo "Diff:"
  diff test23.sml.test test23.sml.expected
  exit 1
fi