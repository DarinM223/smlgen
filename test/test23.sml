type a = b
type b = a

datatype foo = Foo of a

datatype bar = Bar of x
withtype x = y
and y = x