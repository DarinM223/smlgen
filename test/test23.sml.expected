structure Component1 = struct datatype bar = Bar of y end
structure Component2 = struct datatype foo = Foo of b end
type a = b
type b = a
datatype foo = datatype Component2.foo
datatype bar = datatype Component1.bar
